src/test/unit/BLSSignatureCheckerUnit.t.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "../../contracts/middleware/BLSSignatureChecker.sol";
+import "../utils/BLSMockAVSDeployer.sol";
+
+contract BLSSignatureCheckerUnitTests is BLSMockAVSDeployer {
+    using BN254 for BN254.G1Point;
+
+    BLSSignatureChecker blsSignatureChecker;
+
+    function setUp() virtual public {
+        _setUpBLSMockAVSDeployer();
+
+        blsSignatureChecker = new BLSSignatureChecker(registryCoordinator);
+    }
+
+    function testBLSSignatureChecker_SingleQuorum_Valid(uint256 pseudoRandomNumber) public { 
```

Comment:
it would be *awesome* if each test could have a few points like:
one sentence what this is supposed to do
things it should check
any assumed preconditions, or at least some of the important ones

some of these could also go inside the test; I just find it makes it much easier to understand

Reply:
added a couple for the validity non trivial tests